### PR TITLE
⚡️ Rewrite the Contributions Chapter

### PR DESCRIPTION
**Pull request by sandervdo**
Based on the other contributions to this chapter (#36) and the discussions that have been going on (#20, #3) I've rewritten the chapter.

- I've tried to trim down the part of VSCode so that it would not be 90% VSCode and 10% others.
- Introduced emoji's inline as proposed by @sveningsonrobin originally in #20.
- Changed into a narrative format.
- Rewrote the Kibana contributions as well.

*Everyone* is free to throw feedback, this is just a try-out to see whether this is an approach we should continue.

@delftswa2017/team-kibana, @delftswa2017/team-vscode please verify that this still represents your work as you intended.

@delftswa2017/team-yarn @delftswa2017/team-junit5 iirc you were also planning / discussing about this chapter, can you take a look too?

Cheers. 🎉

### CODE REVIEW

File: contributions-chapter/chapter.md [hunk 1]
```diff
@@ -1,15 +1,98 @@
 # Contributions for DESOSA 2017
 
-This chapter outlines the conributions that each of the teams in the course were able to make to their open-source project. Contributions have been categorized and are presented below:
+This chapter outlines some of the contributions made by several teams to their open-source project. Contributions have been categorized based on their types. At the end a list of all mentioned Pull Requests will be provided for more information. It should be mentioned that much more Pull Requests were filed during the course, this chapter only sketches a broad overview of the diversity.
```
Comment (vm00z) [inline contributions-chapter/chapter.md]: I always use open source without a dash but I see some people use it with a dash... So what's the right one? 😓

Comment (vm00z) [reply contributions-chapter/chapter.md]: At the end[,] a list of all mentioned Pull Requests [is] provided...

File: contributions-chapter/chapter.md [hunk 2]
```diff
@@ -1,15 +1,98 @@
...
+[VS-bug-2]: https://github.com/Microsoft/vscode/issues/22401
+[VS21859]: https://github.com/Microsoft/vscode/pull/21859
+[VS22743]: https://github.com/Microsoft/vscode/pull/22743
+[@sandy081]: https://github.com/sandy081
 
 ## Technical debt
-- the Kibana team is trying to remove unnecessary usage of the [Lodash](https://lodash.com) library. There are multiple methods in Lodash that can easily be replaced by very similar native ES6 methods. This [pull request](https://github.com/elastic/kibana/pull/10746) removes some of the unnecessary lodash methods.
-- After analyzing Kibana using tools to look for technical debt, some syntax errors were found in the test code. There was an incorrect `.json` file and a quoting issue with a `.sh` file. This [pull request](https://github.com/elastic/kibana/pull/10747) solved these issues.
+
+Another popular topic was technical debt. As part of the course the (amount of) technical debt was analysed using tools like [SonarQube]. Teams could use their findings to improve the projects that they were analysing.
```
Comment (vm00z) [inline contributions-chapter/chapter.md]: using tools, such as SonarQube.

File: contributions-chapter/chapter.md [hunk 3]
```diff
@@ -1,15 +1,98 @@
...
+
+For instance, the Kibana team is trying to remove unnecessary usage of the [Lodash](https://lodash.com) library ([#10746][KI10746] :white_check_mark:). There are multiple methods in Lodash that can easily be replaced by very similar native ES6 methods. These unnecessary lodash methods have been removed in the contribution. Using different tools to look for technical debt the Kibana team found more issues. Some syntax errors were found in the test code. There was an incorrect `.json` file and a quoting issue with a `.sh` file. This was another possibility for a contribution ([#10747][KI10747] :white_check_mark:).
```
Comment (vm00z) [inline contributions-chapter/chapter.md]: ...the Kibana team tried to remove...

**Review by vm00z — DISMISSED** [review]
Spotted a few mistakes / awkward phrasings.

File: contributions-chapter/chapter.md [hunk 4]
```diff
@@ -1,15 +1,98 @@
...
+
+Several teams went through the issue lists of their projects to find bugs to fix.
+An example would be the [bug][VS-bug] in the search functionality of Visual Studio Code:
+when a user searched for two terms in Visual Studio Code, the 'clear results' button was clicked  and alt+up was used to get the last search item, the second last was shown.
+
+To get the last history item the function `showPreviousSearchTerm()` was called in the file `src\vs\workbench\parts\search\browser\searchWidget.ts`. This function is shown in [Snippet 1](#snippet-1)
+
+<div id="snippet-1"></div>
```
Comment (TimvdLippe) [inline contributions-chapter/chapter.md:22]: There should be a new line after each closing tag. It breaks the snippet rendering

File: contributions-chapter/chapter.md [hunk 5]
```diff
@@ -1,15 +1,98 @@
...
+
+[KI10709]: https://github.com/elastic/kibana/pull/10709
+[KI10714]: https://github.com/elastic/kibana/pull/10714
+[KI10715]: https://github.com/elastic/kibana/pull/10715
 
 ## Bug fixes
-- This Kibana [issue](https://github.com/elastic/kibana/issues/9108) depicts a problem in Safari where exporting data would lead to a display of the JSON in the browser instead of a download action (labeled P3: nice to have bug-fixes). Investigation led to finding out it was a [bug](https://bugs.webkit.org/show_bug.cgi?id=102914) in Webkit that has recently been fixed. An upgraded version of Safari (10.1) should thus solve the problem. Workaround possibilities exist, but these are unreliable. Hence, this analysis was posted in the [issue](https://github.com/elastic/kibana/issues/9108) on Github along with a suggesion for the Kibana team to wait for the new version of Safari. For more info on the download library please visit this [link](eligrey/FileSaver.js#12).
+
+Several teams went through the issue lists of their projects to find bugs to fix.
+An example would be the [bug][VS-bug] in the search functionality of Visual Studio Code:
```
Comment (TimvdLippe) [inline contributions-chapter/chapter.md]: (VS-bug)

File: contributions-chapter/chapter.md [hunk 6]
```diff
@@ -1,15 +1,98 @@
...
+	}
+```
+**Snippet 2** - `showPreviousSearchTerm` with the fix applied
+
+To propose this solution for Microsoft Visual Studio Code a pull request was filed ([#21859][VS21859] :white_check_mark:). After signing the CLA the Travis build had
+to be fixed and some more feedback had to be processed. A maintainer ([@sandy081]) found a
+problem with the proposed solution. After resolving this problem the pull request
+was accepted by the project.
+
+Another [bug][VS-bug-2] that has been resolved is a problem with prefix matching within Visual Studio Code. When a user searched for the number '4' in both command palette and the settings editor of Visual Studio Code. The search function did not only match '4', but also matched the letter 'T'. So these search functions matched false positives.
```
Comment (TimvdLippe) [inline contributions-chapter/chapter.md]: (VS-bug-2)

**Review by TimvdLippe — COMMENTED** [review]
I like it. I will try to add Yarn today, if I can get some spare time

**Comment by TimvdLippe** [comment]
There seems to be inconsistencies between the GitHub rendered markdown and the gitbook rendered markdown.

First of all, GitHub breaks when adding manual `<div>` without a newline behind it. In contrast, the emoticons do not work in the gitbook website: 
image

Should we try to fix the rendering of both engines or focus on one? Nonetheless, either issue has to be fixed :stuck_out_tongue:

**Comment by sandervdo** [comment]
@TimvdLippe hmm good catch, I'll try to replace them with their Unicode equivalences. Maybe that works. 😄 The other feedback I'll process later today.

**Comment by TimvdLippe** [comment]
I am fixing several issues as we speak, so pull request incoming!

**Review by vm00z — APPROVED** [review]
:+1:

**Pull request merged**